lib/rules/jest/no-if.js

```diff
@@ -94,3 +106,10 @@ function hasEmptyBody(node) {
     node.arguments[1].body.body.length === 0
   );
 }
+
+function skip(state) {
+  return (
+    state.callExpressionDepth === 0 ||
+    state.blockStatementDepth > state.callExpressionDepth
```

Comment:
I have a little trouble understanding this condition, can you explain it?

Reply:
So the first part is the same as what was there previously if we are not inside of a testing-CallExpression (`it()`, `test()`, etc...), then we don't care about if statements. 

The part after the `||` is new and is trying to determine if we are inside of a BlockStatement that is inside of (but not attached to) a testing-CallExpesssion. Because the opening `{...}` to the CallExpesssion, for example in `it('works'){ //here... }`, will increase both our blockStatementDepth and callExpessionDepth by one, the greater-than check will tell us if we have entered a BlockStatement that is NOT attached to a CallExpression. In other words, if the blockStatementDepth is greater than the callStatementDepth, than we have entered at least one more BlockStatment, while also inside of a testing-CallExpession, which is valid and we want to `skip` (not report) the current ifStatement.

Open to other ways to handle this as well!

Reply:
Does it catch things like defining an in-line function in an it block (const foo = () => bar ? baz : qux)?

Reply:
I think that should be covered already, will add a test and confirm 👍

Reply:
@lemonmade added passing tests for the arrow functions as well.